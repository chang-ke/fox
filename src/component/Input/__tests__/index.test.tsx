import React from 'react';
import {shallow, mount} from 'enzyme';
import {Input} from '../..';

describe('render <Input/>', () => {
  it('test classname exits', () => {
    const wrapper = shallow(<Input />);
    expect(
      wrapper
        .find('input')
        .at(0)
        .hasClass('fox-input')
    ).toBe(true);
  });

  it('test disabled with true of props', () => {
    const wrapper = shallow(<Input disabled={true} />);
    expect(
      wrapper
        .find('input')
        .at(0)
        .hasClass('fox-input-disable')
    ).toBe(true);
  });

  it('test size with large of props', () => {
    const wrapper = shallow(<Input size="large" />);
    expect(
      wrapper
        .find('input')
        .at(0)
        .hasClass('fox-input-lg')
    ).toBe(true);
  });

  it('test value with null', () => {
    const wrapper = mount(<Input value={null} readOnly />);
    expect(
      wrapper
        .find('input')
        .at(0)
        .getDOMNode()
        .getAttribute('value')
    ).toBe('');
  });

  it('test value with "testvalue"', () => {
    const wrapper = mount(<Input value="testvalue" onChange={jest.fn()} />);
    expect(
      wrapper
        .find('input')
        .at(0)
        .getDOMNode()
        .getAttribute('value')
    ).toBe('testvalue');
  });

  it('test focus', () => {
    const wrapper = shallow(<Input />);
    wrapper
      .find('input')
      .at(0)
      .simulate('focus');
    expect(wrapper.state('focus')).toBe(true);
  });

  it('test blur', () => {
    const Blur = jest.fn();
    const wrapper = shallow(<Input onBlur={Blur} />);
    wrapper
      .find('input')
      .at(0)
      .simulate('blur');
    expect(Blur).toBeCalled();
  });
});
