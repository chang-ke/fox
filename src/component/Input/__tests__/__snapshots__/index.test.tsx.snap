// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <Input/> renders correctly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {},
    "children": Array [
      Object {
        "attribs": Object {},
        "children": Array [
          Object {
            "attribs": Object {
              "class": "fox-input-addon-wrapper",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "fox-input-wrapper",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "class": "fox-input-fix-wrapper",
                    },
                    "children": Array [
                      Object {
                        "attribs": Object {
                          "class": "fox-input",
                        },
                        "children": Array [],
                        "name": "input",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "class": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "class": undefined,
                        },
                      },
                    ],
                    "name": "span",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": Object {
                      "attribs": Object {
                        "class": "fox-input-outline",
                      },
                      "children": Array [],
                      "name": "div",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": null,
                      "parent": [Circular],
                      "prev": [Circular],
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "class": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "class": undefined,
                      },
                    },
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                    },
                  },
                  Object {
                    "attribs": Object {
                      "class": "fox-input-outline",
                    },
                    "children": Array [],
                    "name": "div",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": Object {
                      "attribs": Object {
                        "class": "fox-input-fix-wrapper",
                      },
                      "children": Array [
                        Object {
                          "attribs": Object {
                            "class": "fox-input",
                          },
                          "children": Array [],
                          "name": "input",
                          "namespace": "http://www.w3.org/1999/xhtml",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "tag",
                          "x-attribsNamespace": Object {
                            "class": undefined,
                          },
                          "x-attribsPrefix": Object {
                            "class": undefined,
                          },
                        },
                      ],
                      "name": "span",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": [Circular],
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "class": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "class": undefined,
                      },
                    },
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                    },
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
            ],
            "name": "span",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
    ],
    "name": "label",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {},
    "x-attribsPrefix": Object {},
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
